llvm/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -912,12 +912,14 @@ llvm::getLoopEstimatedTripCount(Loop *L,
   }
 
   // Return the estimated trip count from metadata unless the metadata is
-  // missing or has no value.
+  // missing or has no value.  Return std::nullopt if it's zero.
```

Comment:
nit: can you expand why (because it's naively counter-intuitive. I'd just copy and paste the .rst text.)